### DISCUSSION TITLE
CreateAnonymousLink?

### DISCUSSION THREAD

**Discussion by johnkberry** [pnp/powershell#281] [Q&A]
Is it possible to access, or would it be possible to implement, the CSOM method CreateAnonymousLink in PnP PowerShell?

**Comment by johnkberry** [comment]
I suspect this is equivalent to the [createLink](https://docs.microsoft.com/en-us/graph/api/driveitem-createlink?view=graph-rest-1.0) action of the Microsoft Graph API, perhaps something like this?

```
POST /sites/{siteId}/drive/items/{itemId}/createLink
Content-type: application/json

{
  "type": "view",
  "scope": "anonymous"
}
```

**Comment by NiuBClass666** [comment]
Using the CSOM method CreateAnonymousLink in Powerhsell 5.1 with credentials(username/ps) was the way we used until the recent major update of PnP Module which obviously also upgraded the CSOM assemblies it includes. Now PnP 2.x needs to run in Powershell 7.2.x or later and it did not have Microsoft.SharePoint.Client.SharePointOnlineCredentials anymore. So we can't use Microsoft.SharePoint.Client.ClientContext to create anonymous link without a credential (Using PsCredential object as a replacement returns 403 Forbidden). Still trying to find an alternative way to do it in powershell 7 through PnP Module.
